ext/ruby_newt/ruby_newt.c

```diff
@@ -1071,13 +1159,15 @@ static VALUE rb_ext_Form_SetWidth(VALUE self, VALUE width)
   return Qnil;
 }
 
-static VALUE rb_ext_Run_Form(VALUE self)
+static VALUE rb_ext_Form_Run(VALUE self)
 {
-  newtComponent form, co;
+  newtComponent form;
+  struct newtExitStruct *es;
 
   Data_Get_Struct(self, struct newtComponent_struct, form);
-  co = newtRunForm(form);
-  return Data_Wrap_Struct(cWidget, 0, 0, co);
+  es = ALLOC(struct newtExitStruct);
+  newtFormRun(form, es);
+  return Data_Wrap_Struct(cExitStruct, 0, 0, es);
```

Comment:
I believe this needs `xfree` or `free` function as the third parameter. Opinion?

> When the object needs to be disposed of, the garbage collector will call the free routine to free it. If you have allocated any memory yourself (for instance, by using Data_Make_Struct), you'll need to pass a free function---even if it's just the standard C library's free routine. For complex structures that you have allocated, your free function may need to traverse the structure to free all the allocated memory.